### PR TITLE
Release 0.1.0

### PR DESCRIPTION
**Pull request by inooid**
Deploying multiple apps with shared environment variables can be tedious to manage. This GitHub Action should help relieve some of the pains by:

- Easily managing env variables using `.env` files
- Substitutes environment variables used within the manifest

It basically appends the given `.env` file variables to your given Cloud Run YAML file.

## Usage

```yml
name: Deploy

on:
  push:
    branches:
      - main
  pull_request:
    branches:
      - main

jobs:
  deploy:
    name: Deploy
    runs-on: ubuntu-latest

    steps:
      - uses: actions/checkout@v4

      # ... all your google auth steps here

      - name: Generate service declaration
        id: app_manifest
        uses: hedge-video/cloud-run-env-action@0.1.0
        with:
          input: ./deploy/production-app.yaml
          env_file: ./deploy/production.env

      # Example of actually deploying the cloud run app
      - name: Deploy app to Cloud Run
        uses: google-github-actions/deploy-cloudrun@v2
        with:
          region: us-east1
          metadata: ${{ steps.app_manifest.outputs.output }}
```

### CODE REVIEW

File: __tests__/__mocks__/job.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  name: my-test-job
+  labels:
+    cloud.googleapis.com/location: ${APP_LOCATION}
+spec:
+  template:
+    metadata:
+      annotations:
+        run.googleapis.com/execution-environment: gen2
+        run.googleapis.com/network-interfaces: '[{"network":"projects/my-test-project/global/networks/my-test-network","subnetwork":"projects/my-test-project/regions/us-east1/subnetworks/us-east1-my-test-network"}]'
+        run.googleapis.com/vpc-access-egress: private-ranges-only
```
Comment (freekdegreef) [inline __tests__/__mocks__/job.yaml]: Maybe better to stay close to default config

File: __tests__/__mocks__/job.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    spec:
+      taskCount: 1
+      template:
+        spec:
+          serviceAccountName: ${APP_SERVICE_ACCOUNT}
+          containers:
+            - name: my-test-job
+              image: ${IMAGE}
+              command: ['php', 'artisan', 'schedule:run']
```
Comment (freekdegreef) [inline __tests__/__mocks__/job.yaml]: Maybe a non laravel example command would be better here?

File: __tests__/__mocks__/service.yaml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  labels:
+    cloud.googleapis.com/location: ${MANIFEST_LOCATION}
+  annotations:
+    run.googleapis.com/ingress: internal-and-cloud-load-balancing
+spec:
+  template:
+    metadata:
+      annotations:
+        run.googleapis.com/network-interfaces: '[{"network":"projects/my-test-project/global/networks/my-test-network","subnetwork":"projects/my-test-project/regions/us-east1/subnetworks/us-east1-my-test-network"}]'
+        run.googleapis.com/vpc-access-egress: all-traffic
```
Comment (freekdegreef) [inline __tests__/__mocks__/service.yaml]: Same here regarding sticking to default config.

File: __tests__/knative.test.js [hunk 4]
```diff
@@ -0,0 +1,340 @@
...
+          template: {
+            spec: {
+              taskCount: 1,
+              template: {
+                spec: {
+                  containers: [
+                    {
+                      name: 'my-test-job',
+                      image: 'my-image',
+                      command: ['php', 'artisan', 'schedule:run'],
```
Comment (freekdegreef) [inline __tests__/knative.test.js]: Another laravel specific command better replaced by a more agnostic example

**Review by freekdegreef — APPROVED** [review]
Couple of small comments but looks good otherwise.

File: __tests__/__mocks__/job.yaml (hunk 1 above)
Comment (inooid) [reply __tests__/__mocks__/job.yaml]: Good call, cleaned up all unnecessary annotations

File: __tests__/__mocks__/job.yaml (hunk 2 above)
Comment (inooid) [reply __tests__/__mocks__/job.yaml]: Seems that the regular Cloud Run docs don't really use a custom command, so I've removed it completely.

**Pull request merged**